### ISSUE TITLE
TIFF Grayscale float image can't be read 

### ISSUE THREAD

**Issue by diorcety** [haraldk/TwelveMonkeys#374]
We can not load Grayscale float tiff image because the library use javax.imageio.ImageTypeSpecifier.Grayscale which not handle this format and throws IllegalArgumentException("Bad value for bits!")

**Comment by haraldk** [comment]
Hi,

Thanks for reporting! 

I'll look into it when I find the time. In the mean time, can you post the full stack trace and attach a sample image? It always makes it easier to spot the issue.

Best regards,

-- 
Harald K

**Comment by diorcety** [comment]
You can find hacky solution here: https://github.com/diorcety/TwelveMonkeys/commit/5304788b954b6faaa28d3e48106e5edea1bd7797 and the image here http://dl.free.fr/htjgdrz54
(only few pixels to the value 1.0)

**Comment by haraldk** [comment]
@diorcety Did you actually add floating point write support to the writer too? Nice. :-)

.k

**Comment by diorcety** [comment]
not really... it's very hacky, but it works in my use case, and maybe not really correct at all

**Comment by haraldk** [comment]
Fixed in latest master.

-- 
Harald K

**Comment by converge20t** [comment]
I was having the same problem so I just checked out and compiled the latest code. Now when I run:

`BufferedImage image = ImageIO.read("greyscale_pic.tif");`

I don't get an exception anymore. However, whenever I try to access a black pixel using this code:

`int blackPixel = image.getRGB(x, y);`

I get an ArrayIndexOutOfBoundsException. I believe there is a problem with the TIFF reader code. I wrote a work-around and test for this exception but the code is not clean. Any ideas?
